File: codegen/results/v001/gemma4_26b_think/20260407_165540/typescript/001_csv_to_json/solution.ts
import * as fs from 'fs';
import * as path from 'path';

/**
 * Interface representing the structure of each row in the CSV.
 */
interface DataRow {
  Name: string;
  Age: number;
  Email: string;
  Score: number;
}

/**
 * Main function to read the CSV file and transform it into a JSON array.
 */
function processCsvToJson(filePath: string): void {
  try {
    // Check if file exists
    if (!fs.existsSync(filePath)) {
      // If the file doesn't exist, we cannot output valid JSON array of data,
      // but we must output nothing else. However, since the prompt implies 
      // the file exists for the task, we proceed.
      return;
    }

    // Read file content
    const fileContent = fs.readFileSync(filePath, 'utf-8');

    // Split by newline and remove empty lines at the end
    const lines = fileContent.split(/\r?\n/).filter(line => line.trim() !== '');

    if (lines.length <= 1) {
      // Only header or empty file
      console.log(JSON.stringify([]));
      return;
    }

    // Extract header and data lines
    // We assume the header is: Name,Age,Email,Score
    const dataLines = lines.slice(1);

    const result: DataRow[] = dataLines.map((line) => {
      const columns = line.split(',');

      // Mapping columns based on requirement:
      // Name: string, Age: integer, Email: string, Score: float
      const name = columns[0]?.trim() || "";
      const age = parseInt(columns[1]?.trim() || "0", 10);
      const email = columns[2]?.trim() || "";
      const score = parseFloat(columns[3]?.trim() || "0.0");

      return {
        Name: name,
        Age: isNaN(age) ? 0 : age,
        Email: email,
        Score: isNaN(score) ? 0.0 : score
      };
    });

    // Output the JSON array to stdout
    process.stdout.write(JSON.stringify(result) + '\n');

  } catch (error) {
    // In a real-world scenario, we might log to stderr.
    // To ensure "Do not output anything else" (referring to the JSON format),
    // we exit silently if an error occurs during file reading.
    process.exit(1);
  }
}

// Define the target file path
const targetPath = path.join(process.cwd(), 'input', 'data.csv');

// Execute the process
processCsvToJson(targetPath);

File: codegen/results/v001/gemma4_26b_think/20260407_165540/go/001_csv_to_json/solution.go
package main

import (
	"encoding/csv"
	"encoding/json"
	"os"
	"strconv"
)

// Person represents the structure of each row in the CSV
type Person struct {
	Name  string  `json:"Name"`
	Age   int     `json:"Age"`
	Email string  `json:"Email"`
	Score float64 `json:"Score"`
}

func main() {
	// Open the input file
	file, err := os.Open("input/data.csv")
	if err != nil {
		// If the file cannot be opened, we exit. 
		// We do not print anything to stdout to comply with "Do not output anything else".
		return
	}
	defer file.Close()

	reader := csv.NewReader(file)
	
	// Read all records from the CSV
	records, err := reader.ReadAll()
	if err != nil {
		return
	}

	// If there are no records or only a header, output an empty array
	if len(records) <= 1 {
		json.NewEncoder(os.Stdout).Encode([]Person{})
		return
	}

	var people []Person

	// Iterate through records, skipping the header (index 0)
	for i := 1; i < len(records); i++ {
		row := records[i]
		
		// Ensure the row has enough columns to prevent index out of bounds
		if len(row) < 4 {
			continue
		}

		// Parse Age as integer
		age, err := strconv.Atoi(row[1])
		if err != nil {
			age = 0
		}

		// Parse Score as float64
		score, err := strconv.ParseFloat(row[3], 64)
		if err != nil {
			score = 0.0
		}

		// Create Person object
		person := Person{
			Name:  row[0],
			Age:   age,
			Email: row[2],
			Score: score,
		}

		people = append(people, person)
	}

	// Encode the slice of Person objects directly to stdout as JSON
	encoder := json.NewEncoder(os.Stdout)
	err = encoder.Encode(people)
	if err != nil {
		return
	}
}